decisions/C-ManagerInterface.md

```diff
@@ -0,0 +1,205 @@
+# Initial C API Proposal
+
+This proposal is scoped to the C API bindings for the minimal 
+`ManagerInterface` methods that have been ported to C++ to date, 
+specifically `identifier()` and `displayName()`. Despite this tight
+restriction in scope there are several design decisions to be addressed.
+
+## Namespacing
+
+In C, exported symbols must be namespaced by prefixing the symbol name
+with some string.
+
+Since our top-level namespace includes a version tag, which is likely to
+change, we must use a macro to ease maintenance, e.g.
+
+```c++
+#define OPENASSETIO_NS(symbol) ...
+```
+such that `OPENASSETIO_NS(MyStruct)` expands to 
+`openassetio_v0_0_MyStruct`.
+
+## Strings
+
+Taking inspiration from [this conversation](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/194#discussion_r794047468)
+we can bundle a string buffer and length in a single C struct for error
+messages and parameters, e.g.
+
+```c++
+typedef struct {
+  const size_t maxSize;
+  char* buffer;
+  size_t usedSize;
+} OPENASSETIO_NS(SimpleString);
+```
+`usedSize` is for optimisation and safety and can technically be removed
+if we're happy to assume null-termination and accept the performance
+implications.
+
```

Comment:
This form, with a length, means that language bindings don't need to measure the string upon receiving it, nice for non-zero-terminated-string languages. If the buffer has an unmeasured zero at the end of the string, it means that it's C/C++/Python friendly as well.

Reply:
Good point about language support, where they may not use null-termination. Updated in a9a2a74c, where I also added a brief note re. UTF-8 and string lengths.